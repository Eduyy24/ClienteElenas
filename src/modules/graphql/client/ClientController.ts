import {useMutation, useQuery} from '@apollo/client';
import { useEffect, useState } from 'react';
import { LOGIN_GQL } from './gql/mutations';
import { CLIENTS_SEARCH_GQL } from './gql/queries';
import ClientModel from './model/ClientModel';

export const getLoginMutation = () => useMutation(LOGIN_GQL);

/**
 * Ejecuta el Query para @var {CLIENTS_SEARCH_GQL}
 * @returns {ClientModel[]}
 */
export const getClients = (): ClientModel[] => {
  const clientsData = useQuery(CLIENTS_SEARCH_GQL)
  const [clients, setClients] = useState(Array<ClientModel>(0))
  
  useEffect(() => {
    if (clientsData.data) {
      setClients(clientsData.data.clientsSearch.results);
    }
  }, [clientsData.data]);

  return clients;
}