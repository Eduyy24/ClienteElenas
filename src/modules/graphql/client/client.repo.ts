import {useMutation} from '@apollo/client';
import { LOGIN_GQL } from './gql/mutations';

export const getLoginMutation = () => useMutation(LOGIN_GQL);