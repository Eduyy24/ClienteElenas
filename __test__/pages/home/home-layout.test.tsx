import React from 'react';
import { fireEvent, render } from '@testing-library/react-native'
import HomeLayout from '../../../src/pages/home/home-layout';
import { MockedProvider } from '@apollo/client/testing';
import QueryMock from '../../../__mocks__/queries.moks';


describe('<HomeLayout />', () => {
  test('Ui del componente', ()=> {
    const wraper = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <HomeLayout />
      </MockedProvider>
    )
    expect(wraper.toJSON()).toMatchSnapshot()
  })

  test('Probar funcionalidad de crear cliente', ()=> {
    const wraper = render(
      <MockedProvider mocks={QueryMock} addTypename={false}>
        <HomeLayout />
      </MockedProvider>
    )
    const btnCreate = wraper.getByText('CREAR')
    fireEvent.press(btnCreate)
    expect(wraper.queryByText('Cerrar')).toBeTruthy();
    expect(wraper.toJSON()).toMatchSnapshot()
  })

  test('Probar funcionalidad de actualizar cliente', async ()=> {
    const wraper = render(
      <MockedProvider mocks={QueryMock} addTypename={false}>
        <HomeLayout />
      </MockedProvider>
    )

    await new Promise(resolve => setTimeout(resolve, 0)); // wait for response
    
    //expect(wraper.toJSON()).toMatchSnapshot()
  })
})