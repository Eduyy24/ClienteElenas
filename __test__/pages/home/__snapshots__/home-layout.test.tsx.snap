// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeLayout /> Probar funcionalidad de crear cliente 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "maxHeight": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontStyle": "italic",
          "fontWeight": "500",
        }
      }
    >
      Clientes Elenas
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 16,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "maxHeight": 40,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#A2A7A8",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        CREAR
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    visible={true}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 0.15,
            "justifyContent": "center",
            "paddingRight": "4%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text>
            Cerrar
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        Crear cliente
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "padding": 16,
          }
        }
      >
        <RCTScrollView
          showsVerticalScrollIndicator={false}
        >
          <View>
            <View
              style={
                Object {
                  "height": 90,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginVertical": 10,
                  }
                }
              >
                Nombre
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A2A7A8",
                    "borderRadius": 15,
                    "height": "40%",
                    "maxHeight": 50,
                    "minHeight": 30,
                    "paddingHorizontal": 5,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "color": "#fff",
                      "flex": 1,
                    }
                  }
                  testID="Nombre"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 90,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginVertical": 10,
                  }
                }
              >
                Apellido
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A2A7A8",
                    "borderRadius": 15,
                    "height": "40%",
                    "maxHeight": 50,
                    "minHeight": 30,
                    "paddingHorizontal": 5,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "color": "#fff",
                      "flex": 1,
                    }
                  }
                  testID="Apellido"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 90,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginVertical": 10,
                  }
                }
              >
                Cédula
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A2A7A8",
                    "borderRadius": 15,
                    "height": "40%",
                    "maxHeight": 50,
                    "minHeight": 30,
                    "paddingHorizontal": 5,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "color": "#fff",
                      "flex": 1,
                    }
                  }
                  testID="Cédula"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 90,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginVertical": 10,
                  }
                }
              >
                Celular
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A2A7A8",
                    "borderRadius": 15,
                    "height": "40%",
                    "maxHeight": 50,
                    "minHeight": 30,
                    "paddingHorizontal": 5,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "color": "#fff",
                      "flex": 1,
                    }
                  }
                  testID="Celular"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 90,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginVertical": 10,
                  }
                }
              >
                Correo electrónico
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A2A7A8",
                    "borderRadius": 15,
                    "height": "40%",
                    "maxHeight": 50,
                    "minHeight": 30,
                    "paddingHorizontal": 5,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "color": "#fff",
                      "flex": 1,
                    }
                  }
                  testID="Correo electrónico"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 90,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginVertical": 10,
                  }
                }
              >
                País
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A2A7A8",
                    "borderRadius": 15,
                    "height": "40%",
                    "maxHeight": 50,
                    "minHeight": 30,
                    "paddingHorizontal": 5,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "color": "#fff",
                      "flex": 1,
                    }
                  }
                  testID="País"
                  underlineColorAndroid="transparent"
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 90,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginVertical": 10,
                  }
                }
              >
                Departamento
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A2A7A8",
                    "borderRadius": 15,
                    "height": "40%",
                    "maxHeight": 50,
                    "minHeight": 30,
                    "paddingHorizontal": 5,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                      }
                    }
                  >
                    <TextInput
                      allowFontScaling={true}
                      onChangeText={[Function]}
                      rejectResponderTermination={true}
                      style={
                        Object {
                          "color": "#fff",
                          "flex": 1,
                        }
                      }
                      testID="Departamento"
                      underlineColorAndroid="transparent"
                      value=""
                    />
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignSelf": "center",
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        testID="buscar_Departamento"
                      >
                        Buscar
                      </Text>
                    </View>
                  </View>
                  <View
                    collapsable={false}
                  >
                    <View />
                    <Modal
                      hardwareAccelerated={false}
                      onRequestClose={[Function]}
                      supportedOrientations={
                        Array [
                          "portrait",
                          "portrait-upside-down",
                          "landscape",
                          "landscape-left",
                          "landscape-right",
                        ]
                      }
                      transparent={true}
                      visible={false}
                    >
                      <View
                        accessible={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <View
                          onLayout={[Function]}
                          style={
                            Object {
                              "backgroundColor": "white",
                              "borderRadius": 4,
                              "left": 8,
                              "opacity": 0,
                              "position": "absolute",
                              "shadowColor": "black",
                              "shadowOffset": Object {
                                "height": 2,
                                "width": 0,
                              },
                              "shadowOpacity": 0.14,
                              "shadowRadius": 2,
                              "top": 8,
                              "transform": Array [],
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "overflow": "hidden",
                              }
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              testID="item0"
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "height": 48,
                                      "justifyContent": "center",
                                      "maxWidth": 248,
                                      "minWidth": 124,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <Text
                                  ellipsizeMode="clip"
                                  numberOfLines={1}
                                  style={
                                    Array [
                                      Object {
                                        "fontSize": 14,
                                        "fontWeight": "400",
                                        "paddingHorizontal": 16,
                                        "textAlign": "left",
                                      },
                                      false,
                                      undefined,
                                    ]
                                  }
                                >
                                  <Text>
                                    
                                  </Text>
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </Modal>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 90,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "marginVertical": 10,
                  }
                }
              >
                Dirección
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A2A7A8",
                    "borderRadius": 15,
                    "height": "40%",
                    "maxHeight": 50,
                    "minHeight": 30,
                    "paddingHorizontal": 5,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "color": "#fff",
                      "flex": 1,
                    }
                  }
                  testID="Dirección"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 40,
                  "marginTop": 24,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#A2A7A8",
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  GUARDAR
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`<HomeLayout /> Ui del componente 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "maxHeight": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontStyle": "italic",
          "fontWeight": "500",
        }
      }
    >
      Clientes Elenas
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "padding": 16,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "maxHeight": 40,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#A2A7A8",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        CREAR
      </Text>
    </View>
  </View>
</View>
`;
